### PR TITLE
MAINT: Update to latest muFFT interface and to SurfaceTopography release

### PR DESCRIPTION
**Pull request by pastewka**
This PR fixes compatibility with the latest releases:
- muFFT-0.9.3
- SurfaceTopography-0.90.0

### CODE REVIEW

**Comment by pastewka** [comment]
There is a bug in muFFT that needs to be fixed before this can be merged. I am also thinking about reverting the convenience interface to the old API, as this one is not really "convenient".

**Comment by pastewka** [comment]
Updated downstream muFFT 0.9.2 which contains a critical bug fix

**Comment by pastewka** [comment]
Hi @AntoineSIMTEK, I fixed the downstream bug in muFFT and this should not be ready to merge (for release 0.90.0). Can you review this?

File: ContactMechanics/FFTElasticHalfSpace.py [hunk 1]
```diff
@@ -160,12 +160,12 @@ def fourier_slice(self):
                          zip(self.fourier_locations,
                              self.nb_fourier_grid_pts)))
 
-                def fft(self, arr):
-                    return np.fft.rfftn(arr.T).T
+                def fft(self, arr, arr_out):
```
Comment (AntoineSIMTEK) [inline ContactMechanics/FFTElasticHalfSpace.py:163]: So we stay with the complicated API in muFFT ?

Comment (pastewka) [reply ContactMechanics/FFTElasticHalfSpace.py:163]: Yes, we stick to this. Till does not want to change back.

**Comment by pastewka** [comment]
@AntoineSIMTEK I only found CONTRIBUTING.md. Is there any other file that is duplicate?

**Pull request merged**